src/App.js

```diff
@@ -41,28 +48,55 @@ class App extends Component {
       ...foundActivity,
       isBookmarked: !foundActivity.isBookmarked,
     }
-    console.log(newObject)
 
     this.setState({
       activities: [...startOfNewArray, newObject, ...endOfNewArray],
     })
   }
+  toggleFilter() {
+    this.setState({
+      filter: !this.state.filter,
+    })
+  }
 
   render() {
     return (
       <Grid>
         <Title>Discover</Title>
-        <List>
-          {this.state.activities.map(activity => {
-            return (
-              <ActivityItem
-                text={activity.activity}
-                isBookmarked={activity.isBookmarked}
-                onBookmark={() => this.bookmark(activity.id)}
-              />
-            )
-          })}
-        </List>
+        {this.state.filter === true ? (
+          <List>
+            {this.state.activities
+              .filter(activity => activity.isBookmarked)
+              .map(activity => {
+                return (
+                  <ActivityItem
+                    text={activity.activity}
+                    isBookmarked={activity.isBookmarked}
+                    onBookmark={() => this.bookmark(activity.id)}
+                  />
+                )
+              })}
+          </List>
+        ) : (
+          <List>
+            {this.state.activities.map(activity => {
+              return (
+                <ActivityItem
+                  text={activity.activity}
+                  isBookmarked={activity.isBookmarked}
+                  onBookmark={() => this.bookmark(activity.id)}
+                />
+              )
+            })}
+          </List>
+        )}
+
+        <Footer>
+          <button onClick={e => this.toggleFilter()}>
+            {' '}
```

Comment:
das Leerzeichen kann man vllt hübscher mit margin oder padding in den Styles lösen...